### PR TITLE
[MR-2786] Draft of document based QA db design

### PR DESCRIPTION
**Pull request by macrael**
## Summary
This is a mermaid diagram supporting the designs for the Document based QA flow. I’ll add the questions based one next, we can discuss here and either merge this or delete it and document the decision. 

#### Related issues
https://qmacbis.atlassian.net/browse/MR-2786

### CODE REVIEW

File: docs/technical-design/database-diagram.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+QuestionRound ||--|| User: "requestedBy"            
+
+QuestionRound ||--|{ Document: "has many"
+HealthPlanPackageTable ||--|{ QuestionRound: "has many"
+
+Document {
+    string name
+    string S3URL
+    string uploader
```
Comment (haworku) [inline docs/technical-design/database-diagram.md:106]: What is uploader?

File: docs/technical-design/database-diagram.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    date requestDate
+    date responseDate
+}
+
+QuestionRound ||--|| User: "requestedBy"            
+
+QuestionRound ||--|{ Document: "has many"
+HealthPlanPackageTable ||--|{ QuestionRound: "has many"
+
+Document {
```
Comment (haworku) [inline docs/technical-design/database-diagram.md:103]: Here's a version of DocumentsTable from a couple of RSP ago

```
id 
name 
s3Url
documentCategory
createdAt
sha
virusScan
```

File: docs/technical-design/database-diagram.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    string givenName
+    string familyName
+    string email
+    string role
+}
+
+User }|--|{ State: "many to many"
+
+QuestionRound {
+    string id
```
Comment (haworku) [inline docs/technical-design/database-diagram.md:90]: Maybe its implicit here but `createdAt` is missing. Assuming we would build in a way that we could have drafts

File: docs/technical-design/database-diagram.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    string pkgID
+    string name
+    string requestedBy  
+    date requestDate
+    date responseDate
```
Comment (haworku) [inline docs/technical-design/database-diagram.md:95]: `lastResponseDate` maybe?  There may be multiple responses down the road.

File: docs/technical-design/database-diagram.md (hunk 1 above)
Comment (macrael) [reply docs/technical-design/database-diagram.md:106]: I changed it to uploadedBy, we want to capture who uploaded this document to display that in the list

File: docs/technical-design/database-diagram.md (hunk 2 above)
Comment (macrael) [reply docs/technical-design/database-diagram.md:103]: I don’t think we need the category or sha for now, those were both considerations for pulling documents out of the proto, I think

File: docs/technical-design/database-diagram.md (hunk 1 above)
Comment (haworku) [reply docs/technical-design/database-diagram.md]: gotcha. It should be `user uploadedBy` I think right? We want to start associating with our UserTable.

**Review by haworku — APPROVED** [review]

**Pull request merged**